rl4co/envs/scheduling/jssp.py

```diff
@@ -0,0 +1,665 @@
+from typing import Optional, Tuple
+
+import torch
+
+from tensordict.tensordict import TensorDict
+from torchrl.data import CompositeSpec, DiscreteTensorSpec, UnboundedContinuousTensorSpec
+
+from rl4co.envs.common.base import RL4COEnvBase
+
+
+class Configs:
+    low = 1
+    high = 99
+    et_normalize_coef = 1000
+    rewardscale = 0.0
+    init_quality_flag = False
+
+
+configs = Configs()
+
+
+class JSSPEnv(RL4COEnvBase):
+    """Job Shop Scheduling Problem (JSSP) environment.
+    As per the definition given in https://arxiv.org/pdf/2010.12367.pdf.
+    The goal is to schedule a set of jobs on a set of machines such that the makespan is minimized.
+    In this variation, the number of operations per job is equal to the number of machines.
+
+    Args:
+
+    Note:
+        -
+    """
+
+    name = "jssp"
+
+    def __init__(self, num_jobs, num_machines, **kwargs):
+        super().__init__(**kwargs)
+        self.batch_size = torch.Size([1])
+        adjacency_spec = DiscreteTensorSpec(
```

Comment:
[Minor] For specs, I think it would be cleaner to move them to a `_make_spec(...)` class as done in the [tutorial](https://pytorch.org/rl/tutorials/pendulum.html#:~:text=def%20_make_spec(self%2C%20td_params)%3A)